test/e2e/sparkapplication_test.go

```diff
@@ -176,8 +209,26 @@ var _ = Describe("Example SparkApplication", func() {
 			key := types.NamespacedName{Namespace: app.Namespace, Name: app.Name}
 			Expect(waitForSparkApplicationCompleted(ctx, key)).NotTo(HaveOccurred())
 
-			By("Checking out driver logs")
+			By("Checking out whether resource requests and limits of driver pod are set")
 			driverPodName := util.GetDriverPodName(app)
+			driverPodKey := types.NamespacedName{Namespace: app.Namespace, Name: driverPodName}
+			driverPod := &corev1.Pod{}
+			Expect(k8sClient.Get(ctx, driverPodKey, driverPod)).NotTo(HaveOccurred())
+			for _, container := range driverPod.Spec.Containers {
+				if container.Name != common.SparkDriverContainerName {
+					continue
+				}
+				if app.Spec.Driver.CoreRequest != nil {
+					Expect(container.Resources.Requests.Cpu().Equal(resource.MustParse(*app.Spec.Driver.CoreRequest))).To(BeTrue())
+				}
+				if app.Spec.Driver.CoreLimit != nil {
+					Expect(container.Resources.Limits.Cpu().Equal(resource.MustParse(*app.Spec.Driver.CoreLimit))).To(BeTrue())
+				}
+				Expect(container.Resources.Requests.Memory).NotTo(BeNil())
+				Expect(container.Resources.Limits.Memory).NotTo(BeNil())
+			}
```

Comment:
I'm curious: was there any specific reason to add this assertion? The request and limit fields on the pods are set via `spark-submit` args and aren't patched by the webhook issue that I believe you raised this PR as part of

Reply:
Yes, the cpu/memory requests/limits are not patched by the webhook. I added these assertions just to ensure that `spark-pi-custom-resouce.yaml` works as expected in setting resource requests/limits correctly.

Reply:
What do you think about removing the `coreLimit` field from the existing example applications and creating a new application that specifies both `coreRequest` and `coreLimit`, creating a new end-to-end test for that, then moving this assertion logic to that test? I feel like that captures the intention better but open to your opinion too obviously.

Reply:
@jacobsalway  Sounds good. I have removed the `coreLimit` field from all the example applications except for `spark-pi-custom-resource.yaml`. For that, I added the `coreRequest` and `coreLimit` fields to it.